consensus/src/runway/mod.rs

```diff
@@ -264,40 +321,6 @@ where
         }
     }
 
-    // TODO: we should return an error and handle it outside
-    fn validate_unit(
-        &self,
-        uu: UncheckedSignedUnit<H, D, MK::Signature>,
-    ) -> Option<SignedUnit<'a, H, D, MK>> {
-        let su = match uu.check(self.keybox) {
-            Ok(su) => su,
-            Err(uu) => {
-                warn!(target: "AlephBFT-runway", "{:?} Wrong signature received {:?}.", self.index(), &uu);
-                return None;
-            }
-        };
-        let full_unit = su.as_signable();
-        if full_unit.session_id() != self.session_id {
-            // NOTE: this implies malicious behavior as the unit's session_id
-            // is incompatible with session_id of the message it arrived in.
-            warn!(target: "AlephBFT-runway", "{:?} A unit with incorrect session_id! {:?}", self.index(), full_unit);
-            return None;
-        }
-        if full_unit.round() > self.store.limit_per_node() {
-            warn!(target: "AlephBFT-runway", "{:?} A unit with too high round {}! {:?}", self.index(), full_unit.round(), full_unit);
-            return None;
-        }
-        if full_unit.creator().0 >= self.n_members.0 {
```

Comment:
I guess this check was removed because `uu.check(self.keybox)` already checks per index?

Reply:
Yup, exactly. Good catch though, I wasn't sure anyone would notice.